src/arena.gd

```diff
@@ -45,6 +51,20 @@ func reset():
 
 
 func _ready():
+	for c in $GridMap.get_used_cells():
+		var item = $GridMap.get_cell_item(c)
+		
+		var mesh
+		if item in items_using_tiles:
+			mesh = pksc_tile.instantiate()
+		elif item in items_using_slopes:
+			mesh = pksc_slope.instantiate()
+			mesh.set_basis($GridMap.get_cell_item_basis(c))
+		
+		if mesh:
+			$Tiles.add_child(mesh)
+			mesh.set_position($GridMap.map_to_local(c))
+	
```

Comment:
this part kinda bothers me. is it not posible to use https://docs.godotengine.org/en/stable/classes/class_gridmap.html#class-gridmap-method-get-meshes ? -- something like: `mesh = $GridMap.get_meshes()[c]` ?

Reply:
I don't know how to narrow it down further to a specific surface, for `GridMap` items that also have walls and not just a floor.

In hindsight, this happens because I modeled them in blender, which does not give me control over the order of faces. Had I scripted in meshes in Godot, I could at least control which surface to create in order.